### PR TITLE
Migrates builder to Alpine JDK 15

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This makes things easier once #3218 is addressed

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -35,11 +35,6 @@ RUN ./mvnw -q --batch-mode -DskipTests --also-make -pl zipkin-server install
 RUN mkdir -p /zipkin && cp zipkin-server/target/zipkin-server-*-exec.jar /zipkin && cd /zipkin && jar xf *.jar && rm *.jar
 RUN mkdir -p /zipkin-slim && cp zipkin-server/target/zipkin-server-*-slim.jar /zipkin-slim && cd /zipkin-slim && jar xf *.jar && rm *.jar
 
-FROM maven:3-openjdk-14-slim as zipkin-builder
```
Comment (codefromthecrypt) [inline docker/Dockerfile:39]: I couldn't see where this layer was used..

Comment (anuraaga) [reply docker/Dockerfile:39]: It's used here in a hook

https://github.com/openzipkin/zipkin/blob/206abbc26c333f974e09a08063995f4a23af0a39/docker/hooks/post_push#L38

It's the only way to have these cache across builds with docker hub that I could find.

Comment (codefromthecrypt) [reply docker/Dockerfile:39]: ic, I was actually wondering why we are explicitly caching in the image when we can do stuff like that in travis and github actions..

something to think about, and don't have a better plan at the moment. it is a pity to have copy/pasta..

**Comment by codefromthecrypt** [comment]
Will leave this open a bit. It would be more coherent to depend on openzipkin/jdk which includes an alpine based JDK plus maven. that way "zipkin-builder" which is cache in nature can keep the same process with no more copy/pasta.

**Review by anuraaga — APPROVED** [review]
Cool

**Comment by codefromthecrypt** [comment]
I will merge once local build finishes.

The first time I need to push the builder update to dockerhub kindof exactly when this merges :D

**Comment by codefromthecrypt** [comment]
ok watching master build...

**Pull request merged**